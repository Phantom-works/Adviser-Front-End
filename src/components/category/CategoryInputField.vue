<template>
  <div class="input-container">
      <input v-model="text" class="rounded-input" placeholder="Enter text here" type="text" />
      <DefaultButton class="button" text="+" @click="AddCategory"/>
  </div>
</template>

<script lang="ts" setup>
import DefaultButton from '@/components/default/DefaultButton.vue'
import { ref } from 'vue'
const emit = defineEmits(['AddCategory'])

const text = ref()
function AddCategory() {
  if (!text.value) {
    alert('type category first')
    return
  }
  emit('AddCategory', text.value)

  text.value = ''
}
</script>

<style scoped>
.input-container {
  display: flex;
  flex-wrap: nowrap;
  flex-direction: row;
  width: 100%;
}

.rounded-input {
  background-color: #333;
  border: none;
  border-radius: 10px;
  color: #fff;
  font-size: 1.5rem;
  padding: 1rem;
  flex: 1;
}

.button{
  font-size: x-large;
}
</style>
