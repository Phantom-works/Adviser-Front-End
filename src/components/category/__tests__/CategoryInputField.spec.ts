import { describe, expect, it } from 'vitest'
import { mount } from '@vue/test-utils'
import CategoryInputField from '../CategoryInputField.vue'
describe('CategoryInputField Snapshots', () => {
  // Arrange

  it('Snapshot text empty ', () => {
    //Arrange
    // Act
    const wrapper = mount(CategoryInputField, {
      props: {}
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text long ', () => {
    //Arrange
    // Act
    const wrapper = mount(CategoryInputField, {
      props: {
        text: 'grayhujdsidfhqhj09fbaspuidn'
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Snapshot text empty ', () => {
    //Arrange
    // Act
    const wrapper = mount(CategoryInputField, {
      props: {
        text: 'uid'
      }
    })
    // Assert
    expect(wrapper.html()).toMatchSnapshot()
  })
})
