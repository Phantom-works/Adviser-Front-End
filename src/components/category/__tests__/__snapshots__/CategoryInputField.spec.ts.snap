// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CategoryInputField Snapshots > Snapshot InputDefault 1`] = `"<div data-v-15dafa4e=\\"\\" class=\\"input-container\\"><input data-v-15dafa4e=\\"\\" class=\\"rounded-input\\" placeholder=\\"Enter text here\\" type=\\"text\\"><button data-v-dec1f997=\\"\\" data-v-15dafa4e=\\"\\" class=\\"button\\">+</button></div>"`;

exports[`CategoryInputField Snapshots > Snapshot text empty  1`] = `"<div data-v-15dafa4e=\\"\\" class=\\"input-container\\"><input data-v-15dafa4e=\\"\\" class=\\"rounded-input\\" placeholder=\\"Enter text here\\" type=\\"text\\"><button data-v-dec1f997=\\"\\" data-v-15dafa4e=\\"\\" class=\\"button\\">+</button></div>"`;

exports[`CategoryInputField Snapshots > Snapshot text empty  2`] = `"<div data-v-15dafa4e=\\"\\" class=\\"input-container\\" text=\\"uid\\"><input data-v-15dafa4e=\\"\\" class=\\"rounded-input\\" placeholder=\\"Enter text here\\" type=\\"text\\"><button data-v-dec1f997=\\"\\" data-v-15dafa4e=\\"\\" class=\\"button\\">+</button></div>"`;

exports[`CategoryInputField Snapshots > Snapshot text long  1`] = `"<div data-v-15dafa4e=\\"\\" class=\\"input-container\\" text=\\"grayhujdsidfhqhj09fbaspuidn\\"><input data-v-15dafa4e=\\"\\" class=\\"rounded-input\\" placeholder=\\"Enter text here\\" type=\\"text\\"><button data-v-dec1f997=\\"\\" data-v-15dafa4e=\\"\\" class=\\"button\\">+</button></div>"`;
